### PR TITLE
fix(r2 workers example): correct status code

### PR DESCRIPTION
**Pull request by Hydrophobefireman**
send 206,200 or 304 for successful responses

### CODE REVIEW

**Comment by vlovich** [comment]
LGTM @deadlypants1973

File: content/r2/examples/demo-worker.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ function parseRange(encoded: string | null): undefined | { offset: number, lengt
     return
   }
 
-  const parts = encoded.split('-')
+  const parts = encoded.split("bytes=")[1]?.split("-") ?? []
```
Comment (lukeed) [inline content/r2/examples/demo-worker.md:21]: this is fine for an example i think, but a `Range` header may send a suffix-only value in this format:

```
Range: <unit>=-<suffix>
```

> [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range#syntax)

So splitting on `-` in this case & throwing on `length !== 2` (below) doesn't cover all valid conditions

**Review by lukeed — APPROVED** [review]
As mentioned, this is technically incomplete but probably good enough for a demo snippet. Thanks!

File: content/r2/examples/demo-worker.md (hunk 1 above)
Comment (Hydrophobefireman) [reply content/r2/examples/demo-worker.md:21]: Yeah, even multiple ranges are allowed in a single header and I think at that point we could just have a comment to warn future copy-pasters?

Comment (lukeed) [reply content/r2/examples/demo-worker.md:21]: sure sounds good 👍

**Review by deadlypants1973 — APPROVED** [review]

**Review by vlovich — COMMENTED** [review]
LGTM

**Pull request merged**